### ISSUE TITLE
Dependabot update only first terraform dependency in a single file

### ISSUE THREAD

**Issue by andrewozh** [dependabot/dependabot-core#5150]
Terraform
If there is multiple dependencies located in a single file dependabot will update only first one!

I reproduced a problem here (Check the Pull Requests ):
https://github.com/andrewozh/dependabot-bug-check

**Example:**

Here's a fake .tf file with a two modules! 
0.42.1 version is available here

```
module "first-dep" {
  source  = "cloudposse/elasticache-redis/aws"
  version = "0.42.0"

  name = "it supposed to be updated by dependabot"
}

module "second-dep" {
  source  = "cloudposse/elasticache-redis/aws"
  version = "0.42.0"

  name = "it supposed not to be updated by dependabot"

}
```

I use github-native dependabot, here's my `.github/dependabot.yaml`

```
version: 2
updates:
  - package-ecosystem: "terraform"
    directory: "/"
    schedule:
      interval: "daily"
```

**Expected:**

New version of dependency causes dependabot to create a PRs on updating all dependencies, no matter they located in a single file or not

**Reality:**

Got PR only for first dependency in a file

**Comment by dwc0011** [comment]
TL/DR;  the provider value is used as the dependency name, ie not unique resulting in only one dependency being created
Workaround: (if possible) use git url for source values instead, that appears to be working fine

I believe this is due to the naming for provider sources not being unique, the provider value is used as the name for the dependency as set in [ terraform/file parser.rb](https://github.com/dependabot/dependabot-core/blob/57136896e10834f87a647f99b6d7601e95a45a08/terraform/lib/dependabot/terraform/file_parser.rb#L79) 

Since its name is not unique it is found as an existing dependency and is not added again, thus will only process once. [base/dependency_set](https://github.com/dependabot/dependabot-core/blob/57136896e10834f87a647f99b6d7601e95a45a08/common/lib/dependabot/file_parsers/base/dependency_set.rb#L26)

In order to fix this and add a unique PR for each module referencing the same provider, the naming must change as well as the regex in the [terraform/file_updater.rb method update_registry_declaration](https://github.com/dependabot/dependabot-core/blob/57136896e10834f87a647f99b6d7601e95a45a08/terraform/lib/dependabot/terraform/file_updater.rb#L88) 

Let me know if you are able to address this or not, I am currently working another unrelated task but also within the terraform module and can look at this after that if you are unable to fix this.

**Comment by dwc0011** [comment]
This test is not doing what you think, but first in order to re-run dependabot you need to close any open branches and PRs for modules you have previously run.  While doing testsing, do not merge the branch, just close the branch after closing the PR.

```module "first-dep" {
  source  = "cloudposse/elasticache-redis/aws"
  version = "0.42.0"
}

module "second-dep" {
  source  = "git@github.com:cloudposse/terraform-aws-elasticache-redis.git"
  version = "0.42.0"
}
```

The github link notation does not use the version only terraform registry format does (see [instructions ](https://www.terraform.io/language/modules/sources#github)you linked above from terraform)

That notation says use the latest version of terraform-aws-elasticache-redis.git as evidenced in your log

```
  proxy | 2022/05/19 08:55:38 [023] 200 https://github.com:443/cloudposse/terraform-aws-elasticache-redis.git/info/refs?service=git-upload-pack
updater | INFO <job_369073179> Latest version is 1299a3f6efcc22710b6e418cfeaa6ddd85f36be2
```
You can download the refs file [here](https://github.com:443/cloudposse/terraform-aws-elasticache-redis.git/info/refs?service=git-upload-pack) to verify that 1299a3f6efcc22710b6e418cfeaa6ddd85f36be2 does belong to the latest tag 0.42.1 or to the [release of 0.42.1 on github](https://github.com/cloudposse/terraform-aws-elasticache-redis/releases) and reference the commit id to the left

In order to test the workaround you would need to use this

```
module "first-dep" {
  source  = "cloudposse/elasticache-redis/aws"
  version = "0.42.0"
}

module "second-dep" {
  source  = "git@github.com:cloudposse/terraform-aws-elasticache-redis.git?ref=tags/0.42.0"
}
```

alternatively just use the git notation for both until the bug is fixed

```
module "first-dep" {
  source  = "git@github.com:cloudposse/terraform-aws-elasticache-redis.git?ref=tags/0.42.0"
}

module "second-dep" {
  source  = "git@github.com:cloudposse/terraform-aws-elasticache-redis.git?ref=tags/0.42.0"
}
```

**Comment by andrewozh** [comment]
@dwc0011 it work, thanks!
can i expect to get info in this issue when bug will be fixed?

**Comment by dwc0011** [comment]
> @dwc0011 it work, thanks! can i expect to get info in this issue when bug will be fixed?

Well, I do not work for github nor am I a member of the dependabot team, I just contribute as I have cycles.  That said when a PR is opened it should be linked to this issue and when the PR is satisfied it should also update this ticket and you would be notified.

**Comment by AtzeDeVries** [comment]
I think this might be related https://github.com/dependabot/dependabot-core/issues/5128

**Comment by szemek** [comment]
Hmm, it looks like a pull request I prepared https://github.com/dependabot/dependabot-core/pull/5786 might address this issue.

**Comment by jeffwidman** [comment]
@andrewozh / @szemek should this be closed now that #5786 is merged / deployed? Be great if someone can verify the fix works before we close this.

**Comment by andrewozh** [comment]
@jeffwidman I check it on my temp repo and it works good 🥳
https://github.com/andrewozh/dependabot-bug-check/pull/21/files
@szemek thanks a lot for your work 👍

**Comment by jeffwidman** [comment]
Awesome thanks for letting us know
